`make carthage_package` fails with architectures error

When running `make carthage_package` with Xcode 13.4.1, Carthage 0.38.0 on an Apple Silicon Mac (having first run `make update_carthage_dependencies` as described in `CONTRIBUTING.md`), I get the following error:

    Building and archiving…
    # https://github.com/Carthage/Carthage#archive-prebuilt-frameworks-into-one-zip-file
    # From `carthage help build` we are told that `--archive` implies `--no-skip-current`.
    ./Scripts/carthage-with-workaround-for-issue-3019.sh build --archive --no-use-binaries
    *** xcodebuild output can be found in /var/folders/q2/88t_3skx1rn0gn9znm57tc3c0000gn/T/carthage-xcodebuild.pQGJrT.log
    *** Building scheme "Aspects-tvOS" in Aspects.xcodeproj
    A shell task (/usr/bin/xcrun lipo -create /Users/lawrence/Library/Caches/org.carthage.CarthageKit/DerivedData/13.4.1_13F100/Aspects/1.4.2-ws1/Build/Intermediates.noindex/ArchiveIntermediates/Aspects-tvOS/IntermediateBuildFilesPath/UninstalledProducts/appletvos/Aspects.framework/Aspects /Users/lawrence/Library/Caches/org.carthage.CarthageKit/DerivedData/13.4.1_13F100/Aspects/1.4.2-ws1/Build/Products/Release-appletvsimulator/Aspects.framework/Aspects -output /Users/lawrence/code/ably/ably-cocoa/Carthage/Build/tvOS/Aspects.framework/Aspects) failed with exit code 1:
    fatal error: /Applications/Xcode-13.4.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: /Users/lawrence/Library/Caches/org.carthage.CarthageKit/DerivedData/13.4.1_13F100/Aspects/1.4.2-ws1/Build/Intermediates.noindex/ArchiveIntermediates/Aspects-tvOS/IntermediateBuildFilesPath/UninstalledProducts/appletvos/Aspects.framework/Aspects and /Users/lawrence/Library/Caches/org.carthage.CarthageKit/DerivedData/13.4.1_13F100/Aspects/1.4.2-ws1/Build/Products/Release-appletvsimulator/Aspects.framework/Aspects have the same architectures (arm64) and can't be in the same fat output file
    
    Building universal frameworks with common architectures is not possible. The device and simulator slices for "Aspects" both build for: arm64
    Rebuild with --use-xcframeworks to create an xcframework bundle instead.
    make: *** [carthage_package] Error 1(`make update_carthage_dependencies` works fine, though.)



┆Issue is synchronized with this [Jira Uncategorised](https://ably.atlassian.net/browse/SDK-2589) by [Unito](https://www.unito.io)

--- a/Scripts/carthage-with-workaround-for-issue-3019.sh
+++ b/Scripts/carthage-with-workaround-for-issue-3019.sh
@@ -8,9 +8,8 @@ set -euo pipefail
 xcconfig=$(mktemp /tmp/static.xcconfig.XXXXXX)
 trap 'rm -f "$xcconfig"' INT TERM HUP EXIT
 
-# For Xcode 12 and 13 make sure EXCLUDED_ARCHS is set to arm architectures otherwise
+# For Xcode 13 make sure EXCLUDED_ARCHS is set to arm architectures otherwise
 # the build will fail on lipo due to duplicate architectures.
-echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1200 = arm64 arm64e armv7 armv7s armv6 armv8' >> $xcconfig
 echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1300 = arm64 arm64e armv7 armv7s armv6 armv8' >> $xcconfig
 echo 'EXCLUDED_ARCHS = $(inherited) $(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_$(EFFECTIVE_PLATFORM_SUFFIX)__NATIVE_ARCH_64_BIT_$(NATIVE_ARCH_64_BIT)__XCODE_$(XCODE_VERSION_MAJOR))' >> $xcconfig
 

--- a/Scripts/carthage-with-workaround-for-issue-3019.sh
+++ b/Scripts/carthage-with-workaround-for-issue-3019.sh
@@ -11,6 +11,7 @@ trap 'rm -f "$xcconfig"' INT TERM HUP EXIT
 # For Xcode 13 make sure EXCLUDED_ARCHS is set to arm architectures otherwise
 # the build will fail on lipo due to duplicate architectures.
 echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_x86_64__XCODE_1300 = arm64 arm64e armv7 armv7s armv6 armv8' >> $xcconfig
+echo 'EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_simulator__NATIVE_ARCH_64_BIT_arm64__XCODE_1300 = arm64 arm64e armv7 armv7s armv6 armv8' >> $xcconfig
 echo 'EXCLUDED_ARCHS = $(inherited) $(EXCLUDED_ARCHS__EFFECTIVE_PLATFORM_SUFFIX_$(EFFECTIVE_PLATFORM_SUFFIX)__NATIVE_ARCH_64_BIT_$(NATIVE_ARCH_64_BIT)__XCODE_$(XCODE_VERSION_MAJOR))' >> $xcconfig
 
 export XCODE_XCCONFIG_FILE="$xcconfig"

